app/models/weapon.rb

```diff
@@ -3,5 +3,14 @@ class Weapon < ApplicationRecord
   has_many :characters, through: :character_weapons
   validates :name, :skill, :range, presence: true
   validates :price, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :encum, :hp, :rarity, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than_or_equal_to: 99 }
+  validates :encum, :hp, :rarity, :default_damage, :default_crit, numericality: { only_integer: true, greater_than_or_equal_to: 0, less_than_or_equal_to: 99 }
+
+  scope :order_by_rarity, -> { order(rarity: :desc, price: :desc) }
```

Comment:
You should probably break this into two scopes then, the name is not clear enough to what the scope is actually doing:

```ruby
scope :order_by_rarity, -> { order(rarity: :desc }
scope :order_by_price, -> { order(price: :desc) }
```

Then you can call it like:

```ruby
Weapon. order_by_rarity. order_by_price
```